pkg/cloudprovider/gce/gce.go

```diff
@@ -421,6 +464,57 @@ func (gce *GCECloud) GetNodeResources(name string) (*api.NodeResources, error) {
 	}
 }
 
+func getMetadataValue(metadata *compute.Metadata, key string) string {
+	for _, item := range metadata.Items {
+		if item.Key == key {
+			return item.Value
+		}
+	}
+	return ""
```

Comment:
"" is a valid value for a metadata item. Maybe consider a (string, bool) map-style interface for this function?